src/common/function/src/scalars/json/json_path_exists.rs

```diff
@@ -64,32 +78,61 @@ impl Function for JsonPathExistsFunction {
+            (ConcreteDataType::Binary(_), ConcreteDataType::String(_)) => {
                 for i in 0..size {
                     let json = jsons.get_ref(i);
                     let path = paths.get_ref(i);
 
-                    let json = json.as_binary();
-                    let path = path.as_string();
-                    let result = match (json, path) {
-                        (Ok(Some(json)), Ok(Some(path))) => {
-                            let json_path = jsonb::jsonpath::parse_json_path(path.as_bytes());
-                            match json_path {
-                                Ok(json_path) => jsonb::path_exists(json, json_path).ok(),
-                                Err(_) => None,
-                            }
+                    let json = if let Ok(Some(bytes)) = json.as_binary() {
+                        bytes
+                    } else {
+                        // This should never happen.
+                        return InvalidFuncArgsSnafu {
+                            err_msg: format!("Illegal json binary: {:?}", json),
                         }
-                        _ => None,
+                        .fail();
+                    };
+
+                    let result = if !path.is_null() {
+                        // Since path is not null, we simply
+                        // unwrap it without checking if it's `Null`.
+                        let path = if let Ok(Some(str)) = path.as_string() {
+                            if let Ok(json_path) = jsonb::jsonpath::parse_json_path(str.as_bytes())
+                            {
+                                json_path
+                            } else {
+                                // Should never happen.
+                                return InvalidFuncArgsSnafu {
+                                    err_msg: format!("Illegal json path {:?}", path),
+                                }
+                                .fail();
+                            }
+                        } else {
+                            // Should never happen.
+                            return InvalidFuncArgsSnafu {
+                                err_msg: format!("Illegal json path: {:?}", path),
+                            }
+                            .fail();
+                        };
+
+                        jsonb::path_exists(json, path).ok()
+                    } else {
+                        // Should never reach here since we `match` in advance.
+                        None
                     };
 
                     results.push(result);
                 }
             }
+
+            // Any null args existence causes the result to be NULL.
+            (ConcreteDataType::Null(_), ConcreteDataType::String(_)) => results.push_nulls(size),
+            (ConcreteDataType::Binary(_), ConcreteDataType::Null(_)) => results.push_nulls(size),
+            (ConcreteDataType::Null(_), ConcreteDataType::Null(_)) => results.push_nulls(size),
```

Comment:
According to the sqlness tests, this should be the only change that needs to be done?

Reply:
There are 2 major changes: function signatures and datatype cases.

1. I believe function signature changes matter for accepting null type. I've tested if, with only `Exact(vec![Json, String])`, system will throw error when the signature does not match.
2. Apart from "this" change, I think the error handlers also make sense, while the original simply push a `None` into results when there's a parsing or casting error.